src/app/shared/user/user.service.ts

```diff
@@ -74,14 +80,28 @@ export class UserService {
     }
   }
 
-  getSingleUser(): void {
+  getSingleUser(redirectAfterSuccess: boolean, url?: string): void {
```

Comment:
I suggest just having one parameter, the redirectUrl. Instead of having a parameter that says to honor the second parameter, just have/don't have the url param. I don't think you need it unless there's ever a case where the url is set but you don't want to redirect.